scripts/refine-oauth-token-timeouts.sh

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env bash
+set -eu
+
+# MGDAPI-1016 Configure OAuth access token (session) expiration and token
+# inactivity timeouts for both the internal OAuth server and for selected
+# OAuth clients
+#
+# Note: Only OpenShift user with 'cluster-admin' role/access to the cluster
+# can modify token property definitions. Therefore, this script will work
+# correctly only if run under such a user account!
+
+### Definitions of helper function(s)
+
+# Usage
+function usage() {
+  echo "Set session expiration and token inactivity timeouts for OSD cluster."
+  echo
+  echo "  Number of seconds (without the 's' suffix) are accepted as timeouts."
+  echo
+  echo "Usage:"
+  echo "  $0 [TOKEN INACTIVITY TIMEOUT] [SESSION EXPIRATION TIMEOUT]"
+  echo
+  echo "Default Values:"
+  echo
+  echo "  Minimum value of token inactivity timeout is 300 seconds (5 minutes)."
+  echo "  Defaults to minimum value if not specified."
+  echo
+  echo "  Session expiration timeout defaults to 3000 seconds (50 minutes)"
+  echo "  if not specified (default value of inactivity timeout multiplied"
+  echo "  by ten)."
+  echo
+  echo "Examples:"
+  echo "  # Set session expiration to 10 hours, token inactivity to 30 minutes"
+  echo "  $0 1800 36000"
+  echo "  # Set both session expiration and token inactivity to default values"
+  echo "  $0"
+  echo
+  exit 0
+}
+
+# Display specified message bordered with box of '#' characters
+function show_label() {
+  local message="${1?Please specify heading for the label!}"
+  local length="$(( ${#message} + 8 ))"
+  printf '%0.s#' $(seq 1 $length)
+  echo
+  echo -e "### $message ###"
+  printf '%0.s#' $(seq 1 $length)
+  echo
+}
+
+# Update token expiration or inactivity property on selected OAuth resource
+# (identified by resource type and name) by running 'oc patch' with provided
+# path and new value arguments on that resource
+function oc_patch_token_property() {
+  local resourceType="${1?Please specify OAuth or OAuthClient as a resource!}"
+  local resourceName="${2?Please specify the name of the resource!}"
+  local resourcePath="${3?Please specify the path to the target location!}"
+  local newValue="${4?Please specify the new value of the token property!}"
+
+  # Enumerate supported resource types
+  declare -ra ALLOWED_RESOURCE_TYPES=( "oauth" "oauthclient" )
+
+  # OpenShift namespace to operate at
+  local NAMESPACE="openshift-config"
+
+  # Convert both resource type and name to lowercase
+  resourceType="${resourceType,,}"
+  resourceName="${resourceName,,}"
```

Comment:
Unfortunately this also wouldn't work on Mac.. It can be replaced with
```suggestion
  resourceType=$(awk '{print tolower($0)}' <<< ${resourceType})
  resourceName=$(awk '{print tolower($0)}' <<< ${resourceName})
```

Reply:
Fixed with call of `tr` to avoid time costs of `awk` process launch (not sure if it matters that much).